File: Partner-Core/src/main/java/work/slhaf/partner/module/communication/RollingResult.java
package work.slhaf.partner.module.communication;

import work.slhaf.partner.core.memory.pojo.MemorySlice;
import work.slhaf.partner.core.memory.pojo.MemoryUnit;
import work.slhaf.partner.framework.agent.model.pojo.Message;

import java.util.List;

public record RollingResult(
        MemoryUnit memoryUnit,
        MemorySlice memorySlice,
        List<Message> incrementMessages,
        String summary,
        int rollingSize,
        int retainDivisor
) {
}

File: Partner-Core/src/main/java/work/slhaf/partner/module/communication/RollingResult.kt
package work.slhaf.partner.module.communication

import work.slhaf.partner.core.memory.pojo.MemorySliceSnapshot
import work.slhaf.partner.core.memory.pojo.MemoryUnitSnapshot
import work.slhaf.partner.framework.agent.model.pojo.Message

data class RollingResult(
    val memoryUnit: MemoryUnitSnapshot,
    val memorySlice: MemorySliceSnapshot,
    val rollingSize: Int,
    val retainDivisor: Int,
) {
    val summary: String
        get() = memorySlice.summary ?: ""

    fun incrementMessages(): List<Message> = memoryUnit.messagesOf(memorySlice)
}
